libs/community/langchain_community/document_loaders/parsers/audio.py

```diff
@@ -12,6 +12,144 @@
 logger = logging.getLogger(__name__)
 
 
+class AzureOpenAIWhisperParser(BaseBlobParser):
+    """Transcribe and parse audio files.
+
+    Audio transcription is with Azure OpenAI Whisper model.
+    This is different to the Open AI Whisper parser and requires
+    an Azure OpenAI API Key.
+
+    Args:
+        api_key: Azure OpenAI API key
+        chunk_duration_threshold: minimum duration of a chunk in seconds
+            NOTE: According to the OpenAI API, the chunk duration should be at least 0.1
+            seconds. If the chunk duration is less or equal than the threshold,
+            it will be skipped.
+    """
+
+    def __init__(
```

Comment:
Maybe not for the initial PR, but given this is the Azure version for whisper, I think would be great to also support Entra ID token authentication along with API key. This will probably look like supporting an `azure_ad_token_provider` parameter similar to how it's done in AzureOpenAI: https://github.com/langchain-ai/langchain/blob/44e3e2391c48bfd0a8e6a20adde0b6567f4f43c3/libs/partners/openai/langchain_openai/llms/azure.py#L59

Reply:
I've added an attribute `azure_ad_token` which is set to `None` as default. There's no problem I've found if it gets passed to  `AzureOpenAI` as it's default (None), so it can just be set when need be.